huaweicloud/services/css/resource_huaweicloud_css_es_upgrade_core.go

```diff
@@ -0,0 +1,360 @@
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"upgrade_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"agency": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"indices_backup_check": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"cluster_load_check": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"upgrade_detail": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"start_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"end_time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"agency": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"total_nodes": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"retry_times": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"datastore": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"version": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceCssEsUpgradeCoreCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	clusterID := d.Get("cluster_id").(string)
+	hcV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+	client, err := conf.CssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	// Check whether the cluster status is available.
+	err = checkClusterOperationCompleted(ctx, hcV1Client, clusterID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = createEsUpgradeCore(client, d, clusterID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	expression := "detailList | [?status=='RUNNING'] | [0]"
+	upgradeDetail, err := getEsUpgradeDetail(client, clusterID, expression)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id := utils.PathSearch("id", upgradeDetail, "").(string)
+	if id == "" {
+		return diag.Errorf("not found task ID of the upgrade core")
+	}
+	d.SetId(id)
+
+	// Check whether upgrade core has completed.
+	err = checkUpgradeCoreCompleted(ctx, client, clusterID, d.Id(), d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		goBackErr := removeEsUpgradeCoreImpact(client, d)
```

Comment:
better to replace `go back` with `roll back`
```suggestion
		rollBackErr := removeEsUpgradeCoreImpact(client, d)
```